### ISSUE TITLE
YOLO11 Classifier with a BGR train input?

### ISSUE THREAD

**Issue by Miguel-S-Marques** [ultralytics/ultralytics#20775]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Is there a bug where it doesn't convert to RGB the images during training?

I trained a yolo11n-cls to classify the vehicle color, and if the image input predict was in BGR format (loaded by OpenCV without conversion to RGB) it worked well, and if I converted the input to RGB (via OpenCV cv2.cvtColor) the result was not the expected.
A **real blue car** predicts **blue** if the input is BGR and predicts **red** if the input is RGB.
A **real red car** predicts **red** if the input is BGR and predicts **blue** if the input is RGB.

I had the idea to train a yolo11n-cls with 2 classes (red and blue) and, if I use this code:
```
`import cv2
from ultralytics import YOLO

color_path = r"C:\Users\Miguelmarques\PycharmProjects\PythonProject1\runs\classify\train69\weights\best.pt"

color = YOLO(color_path)

image_path = "blue.png"
image = cv2.imread(image_path)
image =  cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

print(image)


class_results = color(image)[0]

pred_class_id = int(class_results.probs.top1)
pred_class_conf = float(class_results.probs.top1conf)
pred_class_name = color.names[pred_class_id]

print(f"{pred_class_name} ({pred_class_conf:.2%})")`
```

The output is:

```
C:\Users\Miguelmarques\AppData\Local\anaconda3\envs\YOLO11\python.exe C:\Users\Miguelmarques\PycharmProjects\PythonProject1\classifyColor.py 
[[[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 ...

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]]

0: 224x224 red 1.00, blue 0.00, 2.2ms
Speed: 10.1ms preprocess, 2.2ms inference, 0.0ms postprocess per image at shape (1, 3, 224, 224)
red (99.91%)

Process finished with exit code 0
```

And if I comment the line where I convert from BGR to RGB:

`#image =  cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
`

The output is:

```
C:\Users\Miguelmarques\AppData\Local\anaconda3\envs\YOLO11\python.exe C:\Users\Miguelmarques\PycharmProjects\PythonProject1\classifyColor.py 
[[[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 ...

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]]

0: 224x224 blue 0.80, red 0.20, 2.4ms
Speed: 10.2ms preprocess, 2.4ms inference, 0.0ms postprocess per image at shape (1, 3, 224, 224)
blue (80.14%)

Process finished with exit code 0
```

If I change the image to predict to a red image color ("red.png"):
With Conversion BGR -> RGB:
```
C:\Users\Miguelmarques\AppData\Local\anaconda3\envs\YOLO11\python.exe C:\Users\Miguelmarques\PycharmProjects\PythonProject1\classifyColor.py 
[[[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 ...

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]

 [[255   0   0]
  [255   0   0]
  [255   0   0]
  ...
  [255   0   0]
  [255   0   0]
  [255   0   0]]]

0: 224x224 blue 0.80, red 0.20, 2.4ms
Speed: 13.1ms preprocess, 2.4ms inference, 0.0ms postprocess per image at shape (1, 3, 224, 224)
blue (80.14%)

Process finished with exit code 0
```

Without conversion:
```
C:\Users\Miguelmarques\AppData\Local\anaconda3\envs\YOLO11\python.exe C:\Users\Miguelmarques\PycharmProjects\PythonProject1\classifyColor.py 
[[[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 ...

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]

 [[  0   0 255]
  [  0   0 255]
  [  0   0 255]
  ...
  [  0   0 255]
  [  0   0 255]
  [  0   0 255]]]

0: 224x224 red 1.00, blue 0.00, 2.8ms
Speed: 12.5ms preprocess, 2.8ms inference, 0.0ms postprocess per image at shape (1, 3, 224, 224)
red (99.91%)

Process finished with exit code 0
```

### Environment

Ultralytics 8.3.143  Python-3.10.16 torch-2.7.0+cu118 CUDA:0 (NVIDIA GeForce RTX 4070 SUPER, 12282MiB)
Setup complete  (16 CPUs, 31.8 GB RAM, 811.5/930.7 GB disk)

OS                  Windows-10-10.0.22631-SP0
Environment         Windows
Python              3.10.16
Install             pip
Path                C:\Users\Miguelmarques\AppData\Local\anaconda3\envs\YOLO11\lib\site-packages\ultralytics
RAM                 31.82 GB
Disk                811.5/930.7 GB
CPU                 Intel Core(TM) i5-14400F
CPU count           16
GPU                 NVIDIA GeForce RTX 4070 SUPER, 12282MiB
GPU count           1
CUDA                11.8

numpy                1.26.4>=1.23.0
matplotlib           3.10.0>=3.3.0
opencv-python        4.9.0.80>=4.6.0
pillow               11.0.0>=7.1.2
pyyaml               6.0.2>=5.3.1
requests             2.32.3>=2.23.0
scipy                1.15.1>=1.4.1
torch                2.7.0+cu118>=1.8.0
torch                2.7.0+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision          0.22.0+cu118>=0.9.0
tqdm                 4.67.1>=4.64.0
psutil               6.1.1
py-cpuinfo           9.0.0
pandas               2.2.3>=1.1.4
ultralytics-thop     2.0.13>=2.0.0

### Minimal Reproducible Example

.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Miguel-S-Marques, thank you for your detailed bug report and for using Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will review and assist you soon.

We recommend checking the [Docs](https://docs.ultralytics.com/) for general usage, where you’ll find [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions and best practices are covered there.

As this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven’t already. A complete, minimal script and sample images will help us reproduce and debug the issue as quickly as possible.

For custom training or classification questions, sharing all relevant details, including dataset examples and training logs, will help our team provide better guidance. Please also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing

## Upgrade

Ensure you are on the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your report and for helping to improve Ultralytics YOLO! 🦾

**Comment by Y-T-G** [comment]
If the input is a numpy array, it's supposed to be in BGR channel order.

https://docs.ultralytics.com/modes/predict/#inference-sources
